src/database.js

```diff
@@ -69,11 +86,19 @@ export class Database {
     pushToStore(collectionName, records, this.store);
   }
 
-  register(collectionName, factory) {
-    this.factories[collectionName] = factory;
+  register(collectionName, factory, serializer) {
+    this.factories[collectionName] = {factory, serializer};
     this.store[collectionName] = [];
   }
 
+  serialize(record, collectionName) {
+    const serializer = this.serializerFor(collectionName);
+    const records = Array.isArray(record) ? record : [record];
```

Comment:
you can just use `const records = _.castArray(record);` it makes sure that result is array ;0